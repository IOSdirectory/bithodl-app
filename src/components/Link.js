import React, { Component } from 'react';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  link: {
    fontSize: 12,
    fontWeight: '600',
    color: '#FEC98A',
    letterSpacing: 1
  }
});

export default class Link extends Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <Text>
          <StyledText style={[styles.link, this.props.style]}>
            {this.props.children}
          </StyledText>
        </Text>
      </TouchableOpacity>
    );
  }
}

Link.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.any,
  children: PropTypes.node
};
