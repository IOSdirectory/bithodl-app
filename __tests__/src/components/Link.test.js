import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Link from '../../../src/components/Link';

describe('Link', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Link>
        <Text>e80d5b55-cad9-4df9-9272-c0398592411b</Text>
      </Link>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
