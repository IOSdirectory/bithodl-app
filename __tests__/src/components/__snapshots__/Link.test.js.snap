// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#FEC98A",
              "fontSize": 12,
              "fontWeight": "600",
              "letterSpacing": 1,
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        e80d5b55-cad9-4df9-9272-c0398592411b
      </Text>
    </Text>
  </Text>
</View>
`;
